python/src/robyn/modeling/pareto/pareto_optimizer.py

```diff
@@ -0,0 +1,131 @@
+
+
+@dataclass
+class ParetoResult:
+    """
+    A dataclass to hold the results of Pareto optimization.
+
+    Attributes:
+        pareto_solutions (List[str]): List of Pareto-optimal solution IDs.
+        pareto_fronts (int): Number of Pareto fronts considered.
+        result_hyp_param (pd.DataFrame): Hyperparameters of Pareto-optimal solutions.
+        x_decomp_agg (pd.DataFrame): Aggregated decomposition results.
+        result_calibration (Optional[pd.DataFrame]): Calibration results, if applicable.
+        media_vec_collect (pd.DataFrame): Collected media vectors.
+        x_decomp_vec_collect (pd.DataFrame): Collected decomposition vectors.
+        plot_data_collect (Dict[str, Any]): Data for various plots.
+        df_caov_pct_all (pd.DataFrame): Carryover percentage data.
+    """
+
+
+class ParetoOptimizer:
+    """
+    A class to perform Pareto optimization on marketing mix models.
+
+    This class orchestrates the Pareto optimization process, including data preparation,
+    Pareto front calculation, response curve calculation, and plot data preparation.
+
+    Attributes:
+        input_collect (MMMData): Input data for the marketing mix model.
+        output_models (ModelOutputs): Output data from the model runs.
+        response_calculator (ResponseCurveCalculator): Calculator for response curves.
+        carryover_calculator (ImmediateCarryoverCalculator): Calculator for immediate and carryover effects.
+    """
+
+    def __init__(self, input_collect: MMMData, output_models: ModelOutputs):
+        """
+        Initialize the ParetoOptimizer.
+
+        Args:
+            input_collect (MMMData): Input data for the marketing mix model.
+            output_models (ModelOutputs): Output data from the model runs.
+        """
+        self.input_collect = input_collect
+        self.output_models = output_models
+        self.response_calculator = ResponseCurveCalculator(input_collect, output_models)
+        self.carryover_calculator = ImmediateCarryoverCalculator(input_collect, output_models)
+
+    def optimize(
+        self,
+        pareto_fronts: str = "auto",
+        min_candidates: int = 100,
+        calibration_constraint: float = 0.1,
+        quiet: bool = False,
+        calibrated: bool = False,
+    ) -> ParetoResult:
+        """
+        Perform Pareto optimization on the model results.
+
+        Args:
+            pareto_fronts (str): Number of Pareto fronts to consider or "auto".
+            min_candidates (int): Minimum number of candidates to consider.
+            calibration_constraint (float): Constraint for calibration.
+            quiet (bool): Whether to suppress output messages.
+            calibrated (bool): Whether the models are calibrated.
+
+        Returns:
+            ParetoResult: The results of the Pareto optimization.
+        """
+        # Implementation here
+        pass
+
+    def _prepare_data(self) -> Dict[str, pd.DataFrame]:
```

Comment:
It's unclear on what this method does. Can we be specific on what preparations it does?